java/vector/src/main/java/org/apache/arrow/vector/BaseDataValueVector.java

```diff
@@ -74,19 +74,15 @@ public BaseDataValueVector(String name, BufferAllocator allocator) {
   public void clear() {
     if (data != null) {
       data.release();
+      data = allocator.getEmpty();
     }
-    data = allocator.getEmpty();
     super.clear();
   }
 
   @Override
   public void close() {
     clear();
-    if (data != null) {
-      data.release();
-      data = null;
-    }
-    super.close();
+    data = null;
```

Comment:
I would prefer doing this change differently. Maybe by allowing allocator to return an empty buffer even if closed. This is because it makes bugs/issues much easier to understand than getting an NPE.